grpc/federation/error.go

```diff
@@ -38,18 +40,53 @@ func GoWithRecover(eg *errgroup.Group, fn func() (any, error)) {
 	})
 }
 
+type ErrorWithLogAttrs struct {
+	err   error
+	attrs []slog.Attr
+}
+
+func (e *ErrorWithLogAttrs) Error() string {
+	return e.err.Error()
+}
+
+func (e *ErrorWithLogAttrs) Unwrap() error {
+	return e.err
+}
+
+func (e *ErrorWithLogAttrs) GRPCStatus() *grpcstatus.Status {
```

Comment:
May I ask where this is going to be used? 🤔

Reply:
This is used from [grpcstatus.FromError](https://github.com/mercari/grpc-federation/blob/27cf04fb39a1971fd495481a2d48e7add23b22b4/grpc/federation/error.go#L84).
By the way, I originally thought that I only needed to implement Unwrap method, but in fact the error message changed unintentionally, so I implemented this method as well.